pkg/controller/workload/cache/workload_cache.go

```diff
@@ -25,35 +25,38 @@ import (
 	"kmesh.net/kmesh/pkg/nets"
 )
 
-var (
-	WorkloadCache = newWorkloadStore()
-)
+type WorkloadCache interface {
+	GetWorkloadByUid(uid string) *workloadapi.Workload
+	GetWorkloadByAddr(networkAddress NetworkAddress) *workloadapi.Workload
+	AddWorkload(workload *workloadapi.Workload)
+	DeleteWorkload(uid string)
+}
 
 type NetworkAddress struct {
 	Network string
 	Address uint32
 }
 
-type workloadStore struct {
+type cache struct {
 	byUid  map[string]*workloadapi.Workload
 	byAddr map[NetworkAddress]*workloadapi.Workload
 	mutex  sync.RWMutex
 }
 
-func newWorkloadStore() *workloadStore {
-	return &workloadStore{
+func NewWorkloadCache() *cache {
+	return &cache{
 		byUid:  make(map[string]*workloadapi.Workload),
 		byAddr: make(map[NetworkAddress]*workloadapi.Workload),
 	}
 }
 
-func (w *workloadStore) GetWorkloadByUid(uid string) *workloadapi.Workload {
+func (w *cache) GetWorkloadByUid(uid string) *workloadapi.Workload {
```

Comment:
Wouldn't it be better to modify the `w` as well.

Reply:
right, should do